app/code/core/Mage/Catalog/Model/Product/Attribute/Backend/Media.php

```diff
@@ -167,11 +167,19 @@ public function beforeSave($object)
         foreach ($value['values'] as $mediaAttrCode => $attrData) {
             if (array_key_exists($attrData, $newImages)) {
                 $object->setData($mediaAttrCode, $newImages[$attrData]['new_file']);
-                $object->setData($mediaAttrCode . '_label', ($newImages[$attrData]['label'] === null || $newImages[$attrData]['label_use_default']) ? $newImages[$attrData]['label_default'] : $newImages[$attrData]['label']);
+                if ($object->isObjectNew()) {
+                    $object->setData($mediaAttrCode . '_label', $newImages[$attrData]['label']);
+                } else {
+                    $object->setData($mediaAttrCode . '_label', ($newImages[$attrData]['label'] === null || $newImages[$attrData]['label_use_default']) ? $newImages[$attrData]['label_default'] : $newImages[$attrData]['label']);
+                }
```

Comment:
```suggestion
                 $label =  $newImages[$attrData]['label'] === null || !empty($newImages[$attrData]['label_use_default'])
                     ? $newImages[$attrData]['label_default']
                     : $newImages[$attrData]['label']
                $object->setData($mediaAttrCode . '_label', $label);
```

Reply:
look more clean and simply
but this tested in all scenario?

Reply:
No, I didn't test at all. Just solving the error:

> Mage_Eav_Model_Entity_Attribute_Exception: Warning: Undefined array key "label_use_default"  in /shared/httpd/github-openmage/htdocs/app/code/core/Mage/Catalog/Model/Product/Attribute/Backend/Media.php on line 170 in /shared/httpd/github-openmage/htdocs/app/code/core/Mage/Core/Model/Config.php:1478

When assessing `$newImages[$attrData]['label_use_default']`, the result can be 
1. Above warning: missing key 'label_use_default'
2. return value bool 
3. return string '1'|'0'
4. return null

For all the above, `empty()` works nicely.

Reply:
I have to test it with multi store view and new product

Reply:
I had to make some changes to your suggested code because a semicolon was missing, and the case in existsImage also needed to be modified. In the commit, I tried to do a co-authored, but I've never done it before, so I'm not sure if I did it correctly

Reply:
i've tested all scenario